sycl/doc/extensions/IntelGPU/IntelGPUKernelCache.md

```diff
@@ -0,0 +1,54 @@
+# SYCL(TM) Proposal: Intel's Extension to Set Kernel Cache Configuration
+
+
+**NOTE**: Khronos(R) is a registered trademark and SYCL(TM) is a trademark of the Khronos Group, Inc.
+
+A new API will be added to set the cache configuration in the kernel.  This information will be useful to developers tuning specifically for those devices.
+
+This proposal details what is required to provide this API as a SYCL extensions.
+
+## Feature Test Macro ##
+
+The Feature Test Macro will be defined as:
+
+    #define SYCL_EXT_INTEL_KERNEL_CACHE_CONFIG 1
+
+
+# Setting the Kernel Cache Configuration #
+
+The cache configuration of the kernel can be modified to favor large shared local memory (SLM) or large data.  This API gives users a way to do request this change.
+
+The new API is only available when using the Level Zero platform.  The DPC++ default behavior is to expose GPU devices through the Level Zero platform.
+
+``` C++
+enum class kernel_cache_config {
+  cache_large_slm,
+  cache_large_data
+};
+
+cl::sycl::kernel::set_cache_config(const kernel_cache_config);
+```
+
+**Note:** The environment variable SYCL\_ENABLE\_KERNEL\_CACHE\_CONFIG must be set to 1 to modify the kernel cache configuration.
+
+
+## Error Condition ##
+An invalid object runtime error will be thrown if the configuration cannot be modified.
```

Comment:
Is there anything better than "invalid object" for this?

Reply:
This seems under-specified.  Under what circumstances would this error occur?  How can the application avoid getting this error?

Reply:
if the call to  zeKernelSetCacheConfig() fails, an error will be thrown.

Reply:
Looking at the documentation for `zeKernelSetCacheConfig()` here:

https://spec.oneapi.com/level-zero/latest/core/api.html#zekernelsetcacheconfig

It seems like almost all of the errors would occur only due to bugs in the DPC++ implementation.  For example, `ZE_RESULT_ERROR_UNINITIALIZED` would only be returned if DPC++ forgot to initialize the driver, `ZE_RESULT_ERROR_INVALID_NULL_HANDLE` would only be called if DPC++ passed an invalid L0 parameter, etc.

The only exception looks like `ZE_RESULT_ERROR_DEVICE_LOST`, but this seems like a general error that would affect virtually every DPC++ API that touches the device.  For example, I presume this error would also be triggered simply by submitting the kernel to the device.

In any case, errors from `zeKernelSetCacheConfig()` wouldn't be raised from `kernel::set_cache_config()` because (I presume) `kernel::set_cache_config()` won't call that function.  Instead, `kernel::set_cache_config()` would need to remember the cache configuration for the kernel and only call the L0 function later when the kernel is actually submitted to the device.

Reply:
OK.  I removed it.